### PR TITLE
Release/v1.27.0

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Release summary

Version number: v1.27.0

## Release details

AEA:
- Adds auto-generated protobuf `*_pb2.py` and `*_pb2_*.py` files to `coveragerc` ignore template for aea test command.
- Fixes comparison operators ` __eq__`, `__ne__`, `__lt__`, `__le__`, `__gt__`, `__ge__` for multiple classes including `PackageVersion`.
- Adds support to `test packages` command for the `--all` flag and switches default behaviour to only run tests on `dev` packages.
- Fixes miscellaneous issues on base test classes and adds consistency checks via meta classes.

Plugins:
- Updates `open-aea-cli-ipfs` to retry in case an IPFS download fails

Tests:
- Fills coverage gaps on core `aea`
- Fills coverage gaps on multiple packages
- Fills coverage gaps on all plugins

Chores:
- Merges the coverage checks with unit tests and removes the extra test coverage CI run.
- Cleans up release flow.

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [x] Lint and unit tests pass locally and in CI
- [x] I have checked the fingerprint hashes are correct by running (`aea packages lock --check`)
- [x] I have regenerated the latest API docs
- [x] I built the documentation and updated it with the latest changes
- [x] I have added an item in `HISTORY.md` for this release
- [x] I bumped the version number in the `aea/__version__.py` file.
- [x] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

-

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,11 +69,6 @@ test-sub-p:
 	pytest -rfE --doctest-modules aea packages/valory/connections packages/valory/protocols packages/fetchai/connections packages/fetchai/protocols packages/fetchai/skills tests/test_packages/test_$(tdir) --cov=packages.fetchai.$(dir) --cov-report=html --cov-report=xml --cov-report=term-missing --cov-report=term  --cov-config=.coveragerc
 	find . -name ".coverage*" -not -name ".coveragerc" -exec rm -fr "{}" \;
 
-.PHONY: hashes
```
Comment (DavidMinarsch) [inline Makefile:72]: Not needed - use generators. This just means the repo is in a dirty state as docs don't match

File: scripts/RELEASE_PROCESS.md [hunk 2]
```diff
@@ -1,48 +1,40 @@
...
+1. Make sure you have a clean develop virtual environment (`make new_env`). Make sure all tests pass, coverage is at 100% and the local branch is in a clean state (nothing to commit after running `make formatters`, `make code-checks` and `make generators`).
 
-2. Determine the next AEA version
-   Create new release branch named "feature/release-{new-version}, switch to this branch"
-   Run `python scripts/bump_aea_version.py --new-version NEW_VERSION_HERE`. Commit if satisfied.
+2. Determine the next AEA version: Create new release branch named "feature/release-{new-version}, switch to this branch and run `python scripts/bump_aea_version.py --new-version NEW_VERSION_HERE`. Commit if satisfied.
 
 3. Bump plugin versions if necessary by running `python scripts/update_plugin_versions.py --update "PLUGIN_NAME,NEW_VERSION"`. Commit if satisfied.
 
-4. Check the protocols are up-to-date by running `aea generate-all-protocols --check-clean`. Commit if changes occurred. Do the same for test protocols using `aea generate-all-protocols tests/data/packages --check-clean`
```
Comment (DavidMinarsch) [inline scripts/RELEASE_PROCESS.md:12]: This is not needed as it happens as part of `make generators`

File: scripts/RELEASE_PROCESS.md [hunk 3]
```diff
@@ -1,48 +1,40 @@
...
+4. [CURRENTLY SKIPPED] Bump all the packages to their latest versions by running `python scripts/update_package_versions.py`.
 
-5. [CURRENTLY SKIPPED] Bump all the packages to their latest versions by running `python scripts/update_package_versions.py`.
+5. Update the package and dependency hashes, protocols, as well as docs using `make generators`. Commit if changes occurred.
 
-6. Update the package and dependency hashes using `make hashes`. Commit if changes occurred.
```
Comment (DavidMinarsch) [inline scripts/RELEASE_PROCESS.md:16]: same

File: scripts/RELEASE_PROCESS.md [hunk 4]
```diff
@@ -1,48 +1,40 @@
...
+6. Ensure all links are configured `tox -e docs-serve`. Commit if satisfied.
 
-7. Check the package upgrades are correct by running `python -m aea.cli check-packages` and `python scripts/check_package_versions_in_docs.py`. Commit if satisfied.
```
Comment (DavidMinarsch) [inline scripts/RELEASE_PROCESS.md:18]: same

File: docs/upgrading.md [hunk 5]
```diff
@@ -11,6 +11,16 @@ Below we describe the additional manual steps required to upgrade between differ
 
 ## `v1.25.0` to `v1.26.0`
```
Comment (0xArdi) [inline docs/upgrading.md:12]: should it be:
```
## `v1.26.0` to `v1.27.0`
```

File: scripts/RELEASE_PROCESS.md [hunk 6]
```diff
@@ -1,48 +1,40 @@
...
+7. Write release notes and place them in `HISTORY.md`. Add upgrading tips in `upgrading.md`. If necessary, adjust version references in `SECURITY.md`. Commit if satisfied.
 
-8. Check the docs are up-to-date by running `tox -e generate-api-documentation`, `python scripts/check_doc_ipfs_hashes.py --fix` and `python scripts/check_doc_links.py`. Ensure all links are configured `mkdocs serve`. Commit if satisfied.
```
Comment (DavidMinarsch) [inline scripts/RELEASE_PROCESS.md:20]: same

File: scripts/RELEASE_PROCESS.md [hunk 7]
```diff
@@ -1,48 +1,40 @@
...
+8. Run spell checker `./scripts/spell-check.sh`. Run `pylint --disable all --enable spelling ...`. Commit if required.
 
-9. Ensure the signing protocol hash in open-aea is updated: `tests/test_configurations/test_constants.py::test_signing_protocol_hash`
```
Comment (DavidMinarsch) [inline scripts/RELEASE_PROCESS.md:22]: same

File: docs/upgrading.md (hunk 5 above)
Comment (DavidMinarsch) [reply docs/upgrading.md:12]: Yes!

**Pull request merged**